java/client/src/main/java/glide/api/models/commands/RangeOptions.java

```diff
@@ -0,0 +1,250 @@
+/** Copyright GLIDE-for-Redis Project Contributors - SPDX Identifier: Apache-2.0 */
+package glide.api.models.commands;
+
+import static glide.utils.ArrayTransformUtils.concatenateArrays;
+
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+
+public class RangeOptions {
+    public interface ScoreRange {
+        String toArgs();
+    }
+
+    public interface LexRange {
+        String toArgs();
+    }
+
+    public interface IRangeQuery {
```

Comment:
In other languages, they use start interfaces with the letter `I`, but we aren't doing that in Java here.  
Just go with `RangeQuery` as the name of the interface.